src/physics.cpp

```diff
@@ -1160,4 +1166,65 @@ void sample_secondary_photons(Particle& p, int i_nuclide)
   }
 }
 
+// split a particle on the basis of the statistical weight
+// of the current location	
+void split_particle(Particle& p)
+{
+  // skip dead or no energy
+  if (p.E() <= 0 || !p.alive()) return;
+	
+  // Particle's position, weight and energy
+  Position pos  = p.r();
+  double weight = p.wgt();
+  double E = p.E();	      
+
+  // Check if this particle is in the weight weindow mesh and get the mesh bins in each direction
+  int ijk[3] = {0};     // mesh bin in each direction
+  bool in_mesh;  
+  // this is likely expensive 
+  // would like a more generic wrapper in case of cell based
+  // ww's
+  settings::ww_settings->mesh_->get_indices(pos, ijk, &in_mesh);
+  // no vr if not inside the mesh
+  if (!in_mesh) {
+    std::cout << "particle not in mesh" << std::endl;
+    return;
+  }
+
+  // get the paramters 
+  auto params = settings::ww_settings->get_params(p);
+	
+  // if particles weight is above the weight window 
+  // split until they are within the window
+  if (weight > params.upper_weight) {
+    double number = weight/params.upper_weight;  
+    double num = std::min(std::ceil(number), double(params.max_split));
+
+    for (int l = 0; l < num - 1; ++l)  { 
+      p.create_secondary(weight/num, p.u(), p.E(), p.type()); 
+    }
+    p.wgt() = weight/num; 
+    p.wgt_last() = p.wgt();
+	// if the particle weight is below the window  
+  // roulette until the weight is high enough
+  } else if (weight <= params.lower_weight) {  
+    double number = weight/params.survival_weight;
+    if (number < 1.0/params.max_split) {
+      number = 1.0/params.max_split;
+      params.survival_weight = weight/number;
```

Comment:
Why modify the `params`? It seems you use it later, but you could instead just do `p.wgt() = weight/number` later.